python/ray/data/tests/test_dataset.py

```diff
@@ -2450,15 +2450,24 @@ def test_block_builder_for_block(ray_start_regular_shared):
         BlockBuilder.for_block(str())
 
 
-def test_groupby_arrow(ray_start_regular_shared):
-    # Test empty dataset.
-    agg_ds = (
-        ray.data.range_table(10)
-        .filter(lambda r: r["value"] > 10)
-        .groupby("value")
-        .count()
-    )
-    assert agg_ds.count() == 0
+@pytest.mark.parametrize("use_push_based_shuffle", [False, True])
```

Comment:
Nit: You should be able to reduce the context munging and try-finally boilerplate by [parameterizing over a few fixtures](https://docs.pytest.org/en/6.2.x/fixture.html#parametrizing-fixtures) that does this for you, e.g.
```python
@pytest.fixture(params=[True, False])
def use_push_baed_shuffle(request):
    ctx = ray.data.context.DatasetContext.get_current()
    original = ctx.use_push_based_shuffle
    ctx.use_push_based_shuffle = request.param
    yield
    ctx.use_push_based_shuffle = original


def test_groupby_arrow(ray_start_regular_shared, use_push_based_shuffle):
    # Test empty dataset.
    agg_ds = (
        ray.data.range_table(10)
        .filter(lambda r: r["value"] > 10)
        .groupby("value")
        .count()
    )
    assert agg_ds.count() == 0
```